torch_np/_wrapper.py

```diff
@@ -855,8 +784,10 @@ def i0(x: ArrayLike):
 def isscalar(a):
     # XXX: this is a stub
     try:
-        arr = asarray(a)
-        return arr.size == 1
+        from ._ndarray import asarray
+
+        t = asarray(a).get()
+        return t.numel() == 1
```

Comment:
This doesn't feel right. Even if this function is trivial, it should have a split like all the others.

Reply:
OK, done in https://github.com/Quansight-Labs/numpy_pytorch_interop/pull/82/commits/248c02bc8ba2a237471f3b096038642a0101514e

To me, it feels squarely in the cure being worse than a disease territory but hey.

Reply:
So, to put this a bit more in context, making everything uniform in a strict way, even for simple ops like this comes from the following plan:
- Split everything into preprocessing and actual implementation
- Automatise as much of the preprocessing as possible via decorators or any other method. This induces a split API / impl.
- Have a look at how many exceptions do we have left after this. If there's not that many, then that means that we were able to automatise the preprocessing well-enough and the API / impl does not make sense anymore. Then, impl can be moved to the API, again, and have the API part completely removed.

Note that his is just a 3 step program towards a large refactoring. As any refactoring, this could have been done in 1 step, but, if you do that, you may find that step 3 above does not really hold and you cannot really do the refactoring. With this plan, if 3 does not hold, we stay at 2, which is still a better state than the initial one.

Reply:
ISTM we're on the same page here on the big picture. The technical Q going forward is that the `normalizer` decorator allows optional arguments here, which it did not in gh-70. I can either move it over there or leave it here at the top of the 'stack' of gh-70 <- gh-80 <- gh-81 <-gh-82 , whichever make it easier to review.

Reply:
I'm fine leaving it here, unless you really prefer to float it to gh70.

Reply:
Also, I belive you were right. After seeing the machinery necessary to accommodate this one function, I believe that the original solution was good enough. Sorry for the trouble. If you feel like it, feel free to revert.

On a different note, I believe we will want parameters to `normalizer` at some point .Let's discuss some possibilities in a week, once I'm back and I've reviewed gh70.

Reply:
Agreed, we'll need parameters anyway, and there is a solution now, can as well keep it. Let's discuss it when you're back, enjoy your vacation!